src/hooks/useMonitoring.tsx

```diff
@@ -1,12 +1,19 @@
 import {usePrivacyContext} from 'context/privacyContext';
 import {useEffect} from 'react';
 import {monitoring} from 'services/monitoring';
+import {logger} from 'services/logger';
 
 export const useMonitoring = () => {
   const {preferences} = usePrivacyContext();
 
-  useEffect(
-    () => monitoring.enableMonitoring(preferences?.analytics),
-    [preferences?.analytics]
-  );
+  useEffect(() => {
+    const enableMonitoring = async () => {
+      if (preferences?.analytics !== undefined) {
+        await monitoring.enableMonitoring(preferences.analytics);
+      }
+    };
+    enableMonitoring().catch((error: Error) => {
```

Comment:
The monitoring service already checks the `enabled` property and enables the service only if set to true
```suggestion
    monitoring.enableMonitoring().catch((error: Error) => {
```